src/ophyd_async/core/mock_signal_utils.py

```diff
@@ -33,17 +33,16 @@ def set_mock_put_proceeds(signal: Signal[T], proceeds: bool):
 
 
 @asynccontextmanager
-async def mock_puts_blocked(*signals: List[Signal]):
+async def mock_puts_blocked(*signals: Signal):
     for signal in signals:
         set_mock_put_proceeds(signal, False)
     yield
     for signal in signals:
         set_mock_put_proceeds(signal, True)
 
 
-def assert_mock_put_called_with(signal: Signal, value: Any, wait=ANY, timeout=ANY):
-    backend = _get_mock_signal_backend(signal)
-    backend.put_mock.assert_called_with(value, wait=wait, timeout=timeout)
+def get_mock_put(signal: Signal) -> Mock:
```

Comment:
I agree on the docstring point.

I also think the name is confusing (it getting the Mock put of a mock backend rather than the underlying `SoftSignal` put).

It's say `get_backend_put_mock` would be better.

Reply:
I don't like having the word `backend` exposed to the user as nothing else externally facing in this file mentions `backend`, they all use `mock_put` as a phrase so this should match them. Added the docstring though